version: '3.8'
services:
  kod-db:
    image: postgres:14.1-alpine
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - LANG=en_US.UTF-8
      - LC_ALL=en_US.UTF-8
    ports:
      - '5432:5432'
    volumes:
      # - ./database:/var/lib/postgresql/data
      - db:/docker-entrypoint-initdb.d
      - pgdata:/var/lib/postgresql/data
    
    networks:
      - kod
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

  kod-mq:
    image: rabbitmq:3-management-alpine
    container_name: 'rabbitmq'
    ports:
       - 5672:5672
       - 15672:15672
    volumes:
       - ~/.docker-conf/rabbitmq/data/:/var/lib/rabbitmq/
       - ~/.docker-conf/rabbitmq/log/:/var/log/rabbitmq
    networks:
      - kod
  app:
    build: .
    ports:
      - 3001:3001
    depends_on:
      kod-db:
        condition: service_healthy
    networks:
      - kod

volumes:
  kod-db:
  db:
    driver: local
  pgdata: 

networks:
  kod:
    driver: bridge